docs/modules/tls.md

```diff
@@ -321,6 +321,55 @@ The alternative approach is easier for development, and that is to supply the PE
 will store the certificate into the flash chip and turn on verification for that certificate. Subsequent boots of the nodemcu can then
 use `tls.cert.verify(true)` and use the stored certificate.
 
+## tls.cert.auth()
+
+Controls the certificate verification process when the Nodemcu authenticates against the client like when receiving a secure connection.
```

Comment:
" the Nodemcu" feels odd here. Maybe also the term "client" can be ambiguous, no?

Reply:
Changed it. Hope it sounds better to you now. If not I'd be happy for a suggestion.

Reply:
We should give an example how one goes about configuring the Certificate + Key via tls.cert.auth()

e.g.
tls.cert.auth(
[[
-----BEGIN CERTIFICATE-----
CLIENT CERTIFICATE String (PEM file)
-----END CERTIFICATE-----
]]
,
[[
-----BEGIN RSA PRIVATE KEY-----
CLIENT PRIVATE KEY String (PEM file)
-----END RSA PRIVATE KEY-----
]]
)
tls.cert.auth(true)

Similar to the current example for tls.cert.verify()

Reply:
We don't support being a TLS server, so "like when receiving a secure connection" isn't very useful (even if it would, possibly, be correct in an alternate universe in which we *did* support being a TLS server).  However, the `tls.cert.auth` data is used as the TLS **client** key and certificate when the NodeMCU device is making a connection to a remote TLS server.

Reply:
The usecase that I have tested is when the ESP is a MQTT client connecting to a MQTT broker which enforces all three (Cert + Key + CA)

Reply:
@dtran123: Just to be very clear what you mean, because there are as many as six things that might fall into "Cert + Key + CA", I think you mean that you have...

* configured a MQTT broker to enforce some TLS policy, perhaps a simple as "all clients must present TLS certificates signed by `clientCA`" (thing 1)
* installed such a certificate (thing 2) and corresponding private key (thing 3) on the ESP with `tls.cert.auth()`.
* generated a private key (thing 4) and certificate (thing 5) for the MQTT server, with the certificate perhaps signed by `serverCA` (thing 6).
* installed either the broker's certificate or `serverCA` on the ESP with `tls.cert.verify()`.

Is that right?

Reply:
> the `tls.cert.auth` data is used as the TLS **client** key and certificate when the NodeMCU device is making a connection to a remote TLS server.

But how is this different from

## tls.cert.verify()

Controls the certificate verification process when the Nodemcu makes a secure connection.

Reply:
My mosquitto MQTT broker has the following security mosquitto.conf parameters configured: 
cafile, certfile, keyfile. Without a matching set on the ESP, the connection will fail to establish.